src/interfaces/pyo3/py_cover_crypt.rs

```diff
@@ -1,108 +1,61 @@
 use abe_policy::AccessPolicy;
 use cosmian_crypto_core::{
     bytes_ser_de::{Deserializer, Serializable, Serializer},
+    symmetric_crypto::Dem,
     symmetric_crypto::SymKey,
-    KeyTrait,
-};
-use pyo3::{
-    exceptions::{PyException, PyTypeError},
-    prelude::*,
-    types::PyType,
-    PyErr,
 };
+use pyo3::{exceptions::PyTypeError, prelude::*, types::PyType, PyErr};
 
 use crate::{
     api::CoverCrypt as CoverCryptRust,
     interfaces::{
         pyo3::py_abe_policy::Policy,
         statics::{
-            CoverCryptX25519Aes256, EncryptedHeader, MasterSecretKey as MasterSecretKeyRust,
-            PublicKey as PublicKeyRust, SymmetricKey as SymmetricKeyRust,
-            UserSecretKey as UserSecretKeyRust,
+            CoverCryptDem, CoverCryptX25519Aes256, EncryptedHeader,
+            MasterSecretKey as MasterSecretKeyRust, PublicKey as PublicKeyRust,
+            SymmetricKey as SymmetricKeyRust, UserSecretKey as UserSecretKeyRust,
         },
     },
 };
 
+// Vec[u8] return type is converted to a list of bytes instead of
+// the type bytes in Python so we need an explicit conversion
+fn convert_to_pybytes(bytes: &[u8], py: Python) -> PyObject {
+    bytes.into_py(py)
+}
```

Comment:
I am not sure what this conversion does when given a reference on bytes. My guess is that a copy happens. Anyway, I think you always have ownership on the data on the call sites, so you should be able to ask ownership here without problem.
```suggestion
fn convert_into_pybytes(bytes: Vec<u8>, py: Python) -> PyObject {
    bytes.into_py(py)
}
```

Reply:
I used `&[u8]` to avoid calling `as_slice` before `into_py`

Reply:
`into_py()` takes a reference as argument anyway, so there is no need to take ownership. I didn't see it, my mistake.